src/corpus/ui/HowToCite.tsx

```diff
@@ -45,18 +44,83 @@ function ExportButton({
   )
 }
 
+function nameToString(name, format = 'Name, GivenName', initials = false) {
+  const givenName =
+    initials === true
+      ? name.given
+          .split(/[\s,-]+/)
+          .map((n) => `${n[0]}.`)
+          .join(' ')
+      : name.given
+  console.log(givenName, name.given)
+  return format === 'Name, GivenName'
+    ? `${name.family}, ${givenName}`
+    : format === 'GivenName Name'
+    ? `${givenName} ${name.family}`
+    : `${givenName}, ${name.family}`
+}
+
+function namesToString(
+  names: any[],
+  prefix = '',
```

Comment:
I wonder if it would simplify things if we remove the `prefix` parameter (since you only seem to use it once to add "Translated by") and instead add some logic in `CitationText` to check if there are translators, and if so, add the prefix.

Reply:
It is used twice, also in the 'With contributions by' case, so I'm not sure if it would simplify things.